fgpyo/fastx/__init__.py

```diff
@@ -28,18 +28,27 @@
 from contextlib import AbstractContextManager
 from pathlib import Path
 from types import TracebackType
+from typing import TYPE_CHECKING
 from typing import Iterator
 from typing import Optional
 from typing import Set
 from typing import Tuple
 from typing import Type
 from typing import Union
 
-from pysam import FastxFile
-from pysam import FastxRecord
+from fgpyo._optional_dependencies import HAS_PYSAM
+from fgpyo._optional_dependencies import require_pysam
 
+if TYPE_CHECKING:
+    import pysam
+else:
+    if HAS_PYSAM:
+        import pysam
+    else:
+        pysam = None
```

Comment:
The plan would be to just stamp this into all the modules that use `pysam`, and add a call to `require_pysam()` at the top of each function (or class init method) that requires it

Reply:
Yeah that worked a charm